library/src/main/java/com/novoda/downloadmanager/LiteDownloadFileId.java

```diff
@@ -0,0 +1,42 @@
+package com.novoda.downloadmanager;
+
+final class LiteDownloadFileId implements DownloadFileId {
+
+    private final int id;
+
+    LiteDownloadFileId(int id) {
```

Comment:
Could this be moved to a different package? https://github.com/novoda/download-manager/issues/257

Reply:
Can we do package restructuring as a separate PR? Happy to do it today 😄

Reply:
as long as we keep the current classes visibility, I'm happy to move things to packages

Reply:
Perhaps I misunderstood and the #257 doesn't make sense.

For me, I am struggling to understand what `Lite` as a prefix means - in my prior-understanding, I thought it was default implementations of public interfaces which any client could choose to swap out. For this reason, I wanted to move anything with `Lite` to a separate package, to avoid any accessing of internal DLM APIs by a `Lite*` class, to prove these are legit implementations that any client could do with the public API of DLM.

---

In my current understanding, it seems `Lite` is being used as a prefix when it's difficult to name an implementation of an interface. In this cases, I would strongly advise not creating an interface, but simply calling the class by the name that was going to be the interface.

e.g. if you had an interface `Foo` with an implementation `LiteFoo`, just create `Foo` as the class. Keep any methods that would have been on the interface public, and the remaining private/package. This way, when a need arises for a different implementation, an issue can be raised and we can introduce the interface *without having to change the public API* and differentiate the implementation with a name that describes its behaviour, instead of `Lite`.

Reply:
@zegnus this one is for you ^^ I've already had a chat with Ataul 😄

Reply:
There are some swappable components that a client of the library can choose to provide its own custom implementation. In order to provide that we make public interfaces that are the contract that the client must follow.

At the same time, the library provides its internal implementation as out-of-the-box and are not meant by any means to be instantiated by the client. The `Lite...` prefix was a naming attempt to provide this private internal implementation of the interface.

Reply:
Also, it was a design decision to do not provide any visible class to the client but always return contracts as an Interface so that we always code against a contract and we are well aware of it and we are decoupled of a specific implementation.

Reply:
>The Lite... prefix was a naming attempt to provide this private internal implementation of the interface.

this is fine - I thought there were public classes with `Lite` prefix and also there's other classes with an `Internal` prefix.

Reply:
oh, nono :)

Reply:
>also there's other classes with an Internal prefix.

yes yes.